src/Service/Services/MetadataProviders/CosmosSqlMetadataProvider.cs

```diff
@@ -53,9 +55,51 @@ public CosmosSqlMetadataProvider(RuntimeConfigProvider runtimeConfigProvider, IF
                     subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
             }
 
+            foreach (Entity entity in _runtimeConfig.Entities.Values)
+            {
+                CheckFieldPermissionsForEntity(entity);
+            }
+
             _cosmosDb = cosmosDb;
         }
 
+        public void CheckFieldPermissionsForEntity(Entity entity)
+        {
+            foreach (PermissionSetting permission in entity.Permissions)
+            {
+                string role = permission.Role;
+                RoleMetadata roleToOperation = new();
+                object[] Operations = permission.Operations;
+                foreach (JsonElement operationElement in Operations)
+                {
+                    if (operationElement.ValueKind is JsonValueKind.String)
+                    {
+                        continue;
+                    }
+                    else
+                    {
+                        // If not a string, the operationObj is expected to be an object that can be deserialized into PermissionOperation
+                        // object.
+                        if (RuntimeConfig.TryGetDeserializedJsonString(operationElement.ToString(), out PermissionOperation? operationObj, null!)
+                            && operationObj is not null)
+                        {
+                            if (operationObj.Fields is null)
```

Comment:
For expanded compatibility: 
- operationObj.Fields may be null
- operationObj.Fields.Include can have a single entry of wildcard `*` 
- operationObj.Fields.Exclude can be null or empty.

Reply:
Thank you for pointing this out! I reverted the logic to check to see if field is not null instead. :)

Reply:
not doing this.

Reply:
Discussed and not going to support this right now.